src/main/java/dgen/utils/serialization/ParquetSerializer.java

```diff
@@ -0,0 +1,198 @@
+        }
+
+
+        return columnAvro;
+    }
+
+    private String generateAvroSchema(TableConfig tableConfig, Table table) throws IOException {
+        Map<String, Object> tableAvro = new HashMap<>();
+        tableAvro.put("type", "record");
+        tableAvro.put("name", cleanFieldName(table.getAttributeName()));
+
+        List<Map<String, Object>> columnAvros = new ArrayList<>();
+        for (ColumnConfig columnConfig: (List<ColumnConfig>) tableConfig.getObject(TableConfig.COLUMN_CONFIGS)) {
+            Column column = table.getColumn(columnConfig.getInt(ColumnConfig.COLUMN_ID));
+            columnAvros.add(columnToAvro(columnConfig, column));
+        }
+        tableAvro.put("fields", columnAvros);
+
+        return (new ObjectMapper().writeValueAsString(tableAvro));
+    }
+
+    private List<GenericData.Record> generateParquetRecords(Schema parquetSchema, Table table) {
+        List<GenericData.Record> parquetRecords = new ArrayList<>();
+        List<Column> columns = table.getColumns();
+        int numColumns = columns.size();
+
+        // header
+        List<String> attributeNames = new ArrayList<>();
+        List<Iterator<DataType>> columnIterators = new ArrayList<>();
+        for (Column c : columns) {
+            // attribute data for header
+            String attrName = c.getAttributeName();
+            attributeNames.add(cleanFieldName(attrName));
+
+            // the rest of the records
+            Iterator<DataType> it = c.getData().iterator();
+            columnIterators.add(it);
+        }
+
+        boolean moreData = true;
+        while(moreData) {
+            GenericData.Record record = new GenericData.Record(parquetSchema);
+            for (int i = 0; i < numColumns; i++) {
+                Iterator<DataType> it = columnIterators.get(i);
+                String attributeName = attributeNames.get(i);
+                if (! it.hasNext()) {
+                    moreData = false;
+                    continue;
+                }
+                DataType dt = it.next();
+                Object value = dt.value();
+                record.put(attributeName, value);
+            }
+
+            parquetRecords.add(record);
+        }
+
+        parquetRecords.remove(parquetRecords.size() - 1);
+        return parquetRecords;
+    }
+
+    public void writeToParquet(Schema parquetSchema, List<GenericData.Record> recordsToWrite, Path fileToWrite) throws IOException {
+        try (ParquetWriter<GenericData.Record> writer = AvroParquetWriter
+                .<GenericData.Record>builder(fileToWrite)
+                .withSchema(parquetSchema)
+                .withConf(new Configuration())
+                .withCompressionCodec(CompressionCodecName.SNAPPY)
+                .build()) {
+
+            for (GenericData.Record record : recordsToWrite) {
+                writer.write(record);
+            }
+        }
+    }
+
+    public void readFromParquet(Path filePathToRead) throws IOException {
```

Comment:
is this method for testing purposes?

Reply:
It's for testing but I'll leave it in for now. I added a short comment that says its for testing in 4ec4ff75482108855ea85906f3e59ab4525f0b98